tools/buildchallenge.js

```diff
@@ -28,6 +28,7 @@ function parseObjectiveAmount(objective) {
 }
 
 function parseResources(xml) {
+    if( xml.resources === null || xml.resources === undefined ) return [];
```

Comment:
You can use the double equals `==` to create a more simple and readable condition.

Reply:
@2roy999 Could you explain why, please. I thought `===` was always preferable to `==`

Reply:
Well this is the only case which this is ok. The problem with `==` is that it is doing type conversion before doing the actual compare. There is another problem in js that we have actually two type of null/undefined variables which is describing the same thing logically but not in javascript. using `==` can create more simple and readable condition when comparing to null/undefined.

For example the following are equivalence:
`a == null`
`a == undefined`
`a === null || a === undefined`

Reply:
Lets see if I understand. What you are saying is that `a` might be `null` or `a` might be `undefined`. To avoid two tests, we use `==` allowing type conversion, and then we need only test against one or the other. 
Thinking this through further, this seems to mean that if you cast `null` to type `undefined` then it's value becomes `undefined`. And vice versa. Is that correct?